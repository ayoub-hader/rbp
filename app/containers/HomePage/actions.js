import { FILL_FORM } from './constants';

export function fillFormAction(params) {
  return {
    type: FILL_FORM,
    params,
  };
}
