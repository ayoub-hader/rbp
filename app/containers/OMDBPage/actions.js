import {
  LOAD_MOVIES,
  LOAD_MOVIES_SUCCESS,
  LOAD_MOVIES_ERROR,
} from './constants';

export function loadMovies(search) {
  return {
    type: LOAD_MOVIES,
    search,
  };
}

export function moviesLoaded(movies) {
  return {
    type: LOAD_MOVIES_SUCCESS,
    movies,
  };
}

export function moviesLoadingError(error) {
  return {
    type: LOAD_MOVIES_ERROR,
    error,
  };
}
