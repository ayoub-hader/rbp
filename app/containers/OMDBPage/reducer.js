import produce from 'immer';
import {
  LOAD_MOVIES,
  LOAD_MOVIES_SUCCESS,
  LOAD_MOVIES_ERROR,
} from './constants';

export const initialState = {
  loading: false,
  error: false,
  movies: false,
};

/* eslint-disable default-case, no-param-reassign */
const moviesReducer = produce((draft, action) => {
  switch (action.type) {
    case LOAD_MOVIES:
      draft.loading = true;
      draft.error = false;
      draft.movies = false;
      break;
    case LOAD_MOVIES_SUCCESS:
      draft.loading = false;
      draft.error = false;
      draft.movies = action.movies;
      break;
    case LOAD_MOVIES_ERROR:
      draft.loading = false;
      draft.error = action.error;
      break;
  }
}, initialState);

export default moviesReducer;
