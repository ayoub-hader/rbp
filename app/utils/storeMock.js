const storeMock = state => ({
  dispatch: jest.fn(),
  subscribe: jest.fn(),
  getState: jest.fn(() => state),
  replaceReducer: jest.fn(),
  runSaga: jest.fn(),
  createReducer: jest.fn(),
  injectedReducers: jest.fn(),
  injectedSagas: jest.fn(),
});

export default storeMock;
