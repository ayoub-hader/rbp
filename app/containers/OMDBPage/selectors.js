import { createSelector } from 'reselect';
import { initialState } from './reducer';

const selectOmdb = state => state.omdb || initialState;

const makeSelectMovies = createSelector(
  selectOmdb,
  omdbState => omdbState.movies,
);

const makeSelectLoading = createSelector(
  selectOmdb,
  omdbState => omdbState.loading,
);

const makeSelectError = createSelector(
  selectOmdb,
  omdbState => omdbState.error,
);

export { selectOmdb, makeSelectMovies, makeSelectLoading, makeSelectError };
