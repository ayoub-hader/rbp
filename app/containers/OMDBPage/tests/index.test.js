import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import storeMock from 'utils/storeMock';
import { loadMovies } from '../actions';
import OMDBPage from '..';

describe('<OMDBPage />', () => {
  it('should render and match snapshot without movies', () => {
    const store = storeMock({});
    const { container } = render(
      <Provider store={store}>
        <OMDBPage />
      </Provider>,
    );

    expect(container.firstChild).toMatchSnapshot();
    expect(container.querySelector('.movies').childElementCount).toEqual(0);
  });

  it('should render movies list', () => {
    const store = storeMock({
      omdb: {
        movies: [
          {
            Title: 'Jumanji ...',
            Poster: 'N/A',
            Year: '2021',
            imdbID: 'imdb18979',
          },
        ],
      },
    });

    const { container } = render(
      <Provider store={store}>
        <OMDBPage />
      </Provider>,
    );
    expect(container.querySelector('.movies').childElementCount).toEqual(1);
  });

  it('should dispatch load movies action on search click', () => {
    const store = storeMock({});
    const { container } = render(
      <Provider store={store}>
        <OMDBPage />
      </Provider>,
    );

    container.querySelector('input').value = 'jumanji';
    fireEvent.click(container.querySelector('button'));

    expect(store.dispatch).toHaveBeenCalledTimes(1);
    expect(store.dispatch).toHaveBeenCalledWith(loadMovies('jumanji'));
  });
});
