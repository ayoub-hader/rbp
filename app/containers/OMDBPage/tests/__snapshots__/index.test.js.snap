// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OMDBPage /> should render and match snapshot without movies 1`] = `
<div
  class="OMDBPage__Wrapper-m81c9p-0 HgoYX"
>
  <div
    class="SearchInput__Wrapper-sc-1ot3cvd-0 Tcmdk"
  >
    <input
      placeholder="search movies by title"
      type="search"
    />
    <button
      type="button"
    >
      Search
    </button>
  </div>
  <div
    class="movies"
  />
</div>
`;
